src/xrc/src/api.rs

```diff
@@ -235,17 +236,76 @@ fn validate(rate: QueriedExchangeRate) -> Result<QueriedExchangeRate, ExchangeRa
     }
 }
 
+#[derive(Debug, PartialEq, Eq)]
+enum NormalizedTimestampType {
+    /// The timestamp is within the past minute.
+    RequestedOrCurrent,
+    /// The timestamp is within 1-2 minutes ago.
+    Past,
+}
+
+#[derive(Debug)]
+struct NormalizedTimestamp {
+    /// The timestamp in seconds.
+    value: u64,
+    /// Used to determine if the timestamp is a current timestamp or a timestamp from a minute ago.
+    r#type: NormalizedTimestampType,
+}
+
+impl NormalizedTimestamp {
+    /// Creates a new timestamp with the type `RequestedOrCurrent`
+    fn requested_or_current(value: u64) -> Self {
+        Self {
+            value,
+            r#type: NormalizedTimestampType::RequestedOrCurrent,
+        }
+    }
+
+    /// Creates a new timestamp with the type `Past`
+    fn past(value: u64) -> Self {
+        Self {
+            value,
+            r#type: NormalizedTimestampType::Past,
+        }
+    }
+}
+
+/// If the request contains a timestamp, the function returns the normalized requested timestamp.
+/// If the request's timestamp is null, the function returns the current timestamp if no assets are found
+/// to be inflight, OR the normalized timestamp from one minute ago.
```

Comment:
"OR" is a bit confusing. It sounds like the function is allowed to return either.
How about "otherwise"?